hack/lib/util/environment.sh

```diff
@@ -148,7 +148,17 @@ function os::util::environment::setup_tmpdir_vars() {
     HOME="${FAKE_HOME_DIR}"
     export HOME
 
-    mkdir -p  "${BASETMPDIR}" "${LOG_DIR}" "${VOLUME_DIR}" "${ARTIFACT_DIR}" "${HOME}"
+    # ensure that the directories are clean
+    for target in $( ${USE_SUDO:+sudo} findmnt -output TARGET --list ); do
+        if grep -q "${BASETMPDIR}" <<< "${target}"; then
+            ${USE_SUDO:+sudo} umount "${target}"
```

Comment:
The old code only did the `umount` if the `rm -rf ${BASETMPDIR}` had failed—is dropping that condition a deliberate change? Also, the unanchored pattern makes me a little nervous.  We'll probably be fine though because `${BASETMPDIR}` is a fairly long and distinctive path.